src/finch/interface/lazy.py

```diff
@@ -327,7 +337,173 @@ def prod(arr: LazyTensor, dims) -> LazyTensor:
+
+#     if axis is None:
+#         return min(_flatten(arr))
+
+#     # column‑wise minimum
+#     if axis == 0:
+#         return [min(arr[r][c] for r in range(num_rows)) for c in range(num_cols)]
+#     # row‑wise minimum
+
+#     if axis == 1:
+#         return [min(row) for row in arr]
+
+    # TODO: Use operator.min to reduce?
+    return reduce(operator.add, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
+
+def max(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Return the maximum of input array ``arr`` along given axis.
+    """
+    # TODO: Use operator.max to reduce?
+    return reduce(operator.add, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
+
+
+def mean(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Calculates the arithmetic mean of the input array ``arr``.
+    """
+    origin = np.asarray(arr.shape)
+    ele_no = origin[axis]
+    n = lazy(np.full(np.delete(origin, axis), ele_no, dtype=int))
+
+    s = sum(arr, axis=axis, keepdims=keepdims)
+
+    return elementwise(operator.truediv, s, n)
+
+def var(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        correction: int | float = 0.0,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Calculates the variance of the input array ``arr``.
+    """
+    origin = np.asarray(arr.shape)
+    ele_no = origin[axis]
+    n = lazy(np.full(np.delete(origin, axis), ele_no, dtype=int))
+
+    m = mean(arr, axis=axis, keepdims=keepdims)
+    # TODO: Confirm whether we have broadcasting or reshape interface for lazyTensor
+    # v = elementwise(operator.sub, arr, m)
+    # v2 = elementwise(operator.mul, v, v)
+    # return elementwise(operator.truediv, sum(v, axis=axis, keepdims=keepdims), n)
+
+    return reduce(operator.add, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
+
+def std(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        correction: int | float = 0.0,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Calculates the standard deviation of the input array ``arr``.
+    """
+    # TODO: Need sqrt operator?
+    # d = var(arr, axis=axis, keepdims=keepdims, correction=correction)
+    # return elementwise(operator.sqrt, d)
+    return reduce(operator.add, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
+
+def argmin(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Returns the indices of the minimum values along a specified axis.
+    """
+    return reduce(operator.add, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
```

Comment:
see the way we implement in julia: https://github.com/finch-tensor/Finch.jl/blob/c287ea7a920124e4b953aa13ba88a5310e43c1f1/src/interface/lazy.jl#L776-L809

use`range` or `np.ndindex` or `np.ndenumerate` to get lazy indices in python.